src/main/scala/subsystem/HasTiles.scala

```diff
@@ -10,25 +10,47 @@ import freechips.rocketchip.devices.tilelink.{BasicBusBlocker, BasicBusBlockerPa
 import freechips.rocketchip.diplomacy._
 import freechips.rocketchip.diplomaticobjectmodel.logicaltree.{LogicalModuleTree}
 import freechips.rocketchip.interrupts._
-import freechips.rocketchip.tile.{BaseTile, LookupByHartIdImpl, TileParams, HasExternallyDrivenTileConstants, InstantiableTileParams, PriorityMuxHartIdFromSeq}
+import freechips.rocketchip.tile.{BaseTile, LookupByHartIdImpl, TileParams, InstantiableTileParams, MaxHartIdBits}
 import freechips.rocketchip.tilelink._
 import freechips.rocketchip.util._
 
 /** Entry point for Config-uring the presence of Tiles */
 case class TilesLocated(loc: HierarchicalLocation) extends Field[Seq[CanAttachTile]](Nil)
 
+/** Whether a global hart id prefix is going to be dynamically applied to the static tile hart ids.
```

Comment:
```suggestion
/** Whether a global hart id prefix is going to be dynamically applied to the static tile hart ids via external inputs.
```

Reply:
But in the closed source, these prefixes are driven by devices that may or may not be located in the subsystem and bundle-bridged to this node. So I would not necessarily say they are "external".